neural_nets/wrappers/wrappers.py

```diff
@@ -142,11 +142,59 @@ def backward(self, dLdy):
         dLdy *= 1.0 / (1.0 - self.p)
         return self._base_layer.backward(dLdy)
 
+class AlphaDropout(WrapperBase):
+    def __init__(self, wrapped_layer, p):
+        """
+       Alpha Dropout is a `Dropout` that aim at keeping mean and variance to
+       their original values after “alpha dropout”, in order to ensure the
+       self-normalizing property even for “alpha dropout”.
+
+        Parameters
+        ----------
+        wrapped_layer : `layers.LayerBase` instance
+            The layer to apply AlphaDropout to.
+        p : float in [0, 1)
+            The dropout propbability during training(as with Dropout)
+        """
+        super().__init__(wrapped_layer)
+        self.p = p
+        self._init_wrapper_params()
+        self._init_params()
+
+    def _init_wrapper_params(self):
+        self._wrapper_derived_variables = {"dropout_mask": None}
+        self._wrapper_hyperparameters = {"wrapper": "AlphaDropout", "p": self.p}
+
+    def _greaterOrEqual(self,x,y):
+        return x >= y
+
+    def forward(self, X):
+        alpha = 1.6732632423543772848170429916717
+        scale = 1.0507009873554804934193349852946
+        alpha_p = -alpha * scale
+        if self.trainable:
+            scaler = 1.0 - self.p
+            a = ((1 - self.p) * (1 + self.p * alpha_p ** 2)) ** -0.5
+            b = -a * alpha_p * self.p
+
+            keep = self._greaterOrEqual(np.random.uniform(0,1,X.shape))
```

Comment:
Does this run? It looks like you don't pass a `y` value to your `_greaterOrEqual` method.

In general, I don't think you need to create a separate method. A one-liner is still pretty clean (IMO): `np.random.rand(*X.shape) >= self.p`.

Reply:
More generally, this function is *very* similar to the Keras implementation [here](https://github.com/keras-team/keras/blob/ed07472bc5fc985982db355135d37059a1f887a9/keras/layers/noise.py#L144). 

I want to stress that all code in this repo should be your own work, or should properly cite the code it's based on.